lib/CAPI/Dialect/OM.cpp

```diff
@@ -287,6 +288,26 @@ MlirAttribute omReferenceAttrGetInnerRef(MlirAttribute referenceAttr) {
       (Attribute)unwrap(referenceAttr).cast<ReferenceAttr>().getInnerRef());
 }
 
+//===----------------------------------------------------------------------===//
+// IntegerAttr API.
+//===----------------------------------------------------------------------===//
+
+bool omAttrIsAIntegerAttr(MlirAttribute attr) {
+  return unwrap(attr).isa<circt::om::IntegerAttr>();
+}
+
+MlirAttribute omIntegerAttrGetInt(MlirAttribute attr) {
+  return wrap(
+      (Attribute)unwrap(attr).cast<circt::om::IntegerAttr>().getValue());
+}
+
+MlirAttribute omCastIntAttrIfValid(MlirAttribute attr) {
```

Comment:
Should this just be something like `omIntegerAttrGet(MlirAttribute attr)`, and do a direct `cast<mlir::IntegerAttr>(unwrap(attr))`? I think the way you've refactored the Python, we just need a method to get an `om::IntegerAttr` from an `mlir::IntegerAttr`.

Reply:
Thanks @mikeurbach , you are right, simplified the API.